modules/vaos/app/exceptions/vaos_backend_service_exception.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+require 'common/exceptions'
+
+module VAOS
+  module Exceptions
+    class BackendServiceException < Common::Exceptions::BackendServiceException
+      # rubocop:disable Style/MutableConstant, Style/CaseEquality
+      VAOS_ERRORS = {
+        400 => 'VAOS_400',
+        403 => 'VAOS_403',
+        404 => 'VAOS_404',
+        409 => 'VAOS_409A',
+        500..510 => 'VAOS_502'
+      }
+
+      VAOS_ERRORS.default = 'VA900'
+
+      def initialize(env)
+        @env = env
+        key = lookup_key(env.status)
+        super(key, response_values, env.status, env.body)
+      end
+
+      def response_values
+        {
+          detail: detail(@env.body),
+          source: { vamf_url: @env.url, vamf_body: @env.body, vamf_status: @env.status }
+        }
+      end
+
+      private
+
+      def lookup_key(status)
+        VAOS_ERRORS.select { |status_code| status_code === status }.values.first
```

Comment:
NITPICK: Algorithmically speaking this isn't particularly performant and begs the question why use a Hash at all?

Hash lookup is O(n) if the key is known, which it is in most of your cases. A more performant algorithm would take advantage of this fact, and have an else clause for when everything else. But the way you're using it here, it's not really any more performant than having a Case When Else clause (which would actually be a bit more readable and idiomatic).

Reply:
Also, what happens if a status code of 600 is returned? Shouldn't that be interpreted as a 500 too?

Reply:
I think O(n) would represent a scan of the entire hash, I could be wrong here but on average it should be O(1).  That being said it could be different in Ruby.  In either case, the cardinality of the set we're interested in (i.e. the set of http status codes) is fairly small.  In terms of readability I would have to agree, I think the hash table is a bit more concise though.  Also any unrecognized status coded will get mapped to a VA900 response code.  In those cases, you are correct the complexity of those look-ups is O(n), but in our case n=15, so we're talking picoseconds :).

Reply:
And yes, I meant to say that hash lookup is O(1), and that what you've got is essentially O(n). 

You can make it a little bit faster by doing a #find or #detect instead of #select. This will return the first value match as opposed to #select which will iterate across the entire hash even when a match has been found, returning all of them. With #find or #detect you wouldn't need to do #values.first and could instead do #.last since it returns the key value pair as an array.

Reply:
I got rid of the `select`